src/lib/relay/middlewares/timingMiddleware.ts

```diff
@@ -0,0 +1,18 @@
+import { volleyClient } from "lib/utils/volleyClient"
+
+export function timingMiddleware() {
+  return next => req => {
+    const startTime = Date.now()
+    const operation = req.operation.name || "UnknownOperation"
+    return next(req).then(res => {
+      const duration = Date.now() - startTime
+      volleyClient.send({
+        type: "timing",
+        name: "graphql-request-duration",
+        timing: duration,
+        tags: [`operation:${operation}`],
```

Comment:
In addition to the query `name`, could we include the query `id`? That way, we can track performance improvements if we change the query structure.

Reply:
great idea! :+1: